external/radiation/radiation/wrapper_api.py

```diff
@@ -182,3 +188,57 @@ def _rad_compute(self, state: State, time: cftime.DatetimeJulian,) -> Diagnostic
         )
         out = postprocess_out(out)
         return unstack(out, coords)
+
+    def _get_model(
+        self,
+        time: cftime.DatetimeJulian,
+        nz: int,
+        tracer_name_mapping: Mapping[str, str] = TRACER_NAMES_IN_MAPPING,
+    ) -> MutableMapping[Hashable, Any]:
+        model: MutableMapping[Hashable, Any] = {
+            "me": self._comm.rank,
+            "levs": nz,
+            "levr": nz,
+            "nfxr": self._rad_config.nfxr,
+            "ncld": self._rad_config.ncld,
+            "ncnd": self._rad_config.ncnd,
+            "fhswr": self._rad_config.fhswr,
+            "fhlwr": self._rad_config.fhlwr,
+            # todo: why does solar hour need to be one timestep behind time to validate?
+            "solhr": _solar_hour(time - timedelta(seconds=self._timestep)),
+            "lsswr": self._rad_config.lsswr,
+            "lslwr": self._rad_config.lslwr,
+            "imp_physics": self._rad_config.imp_physics,
+            "lgfdlmprad": self._rad_config.lgfdlmprad,
+            "uni_cld": self._rad_config.uni_cld,
+            "effr_in": self._rad_config.effr_in,
+            "indcld": self._rad_config.indcld,
+            "num_p3d": self._rad_config.num_p3d,
+            "npdf3d": self._rad_config.npdf3d,
+            "ncnvcld3d": self._rad_config.ncnvcld3d,
+            "lmfdeep2": self._rad_config.lmfdeep2,
+            "lmfshal": self._rad_config.lmfshal,
+            "sup": self._rad_config.sup,
+            "kdt": self._rad_config.kdt,
+            "do_sfcperts": self._rad_config.do_sfcperts,
+            "pertalb": self._rad_config.pertalb,
+            "do_only_clearsky_rad": self._rad_config.do_only_clearsky_rad,
+            "swhtr": self._rad_config.swhtr,
+            "solcon": self._driver.solar_constant,
```

Comment:
Feels a bit odd that the rest of these are actually config options, while this item is something computed by the code (if I understand correctly). Do we need to include `solcon` in the `_get_model` output?

Reply:
Put another way, can/should we separate out the retrieval of config options that remain the same from the retrieval of things like `solhr` and `solcon` that actually depend on time?

Reply:
We are somewhat constrained by the fact that this is port of the Fortran radiation driver, which when called takes a data structure called `Model` of the `GFS_control_type` that holds a ton of different flags/scalars. Two of those things that `GFS_control_type` contains are `solcon` and `solhr`.

The python port stays true to this, and in general I think we want to have the port and Fortran schemes avoid diverging too much. OTOH we could separate them out in their own arguments since the current way of doing things led to bugs. In terms of the name of this structure, I agree that `model` is not very informative. Maybe `GFS_physics_control` instead?

Reply:
I did make `GFS_physics_control` a typed dataclass that is part of the radiation config, and separated out the time-varying solar arguments from it. I think it's reasonable to try to clean up some of the port's unfortunate features (dicts of untyped objects getting passed around) on a piecemeal basis.